libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreDatabaseManager.m

```diff
@@ -126,6 +134,12 @@ - (id)initGlobalManager
     return self;
 }
 
+- (void)dealloc {
+    if (self.user) {
```

Comment:
Shouldn't this be driven by the implementation of `authManager:willLogoutUser:` in `SalesforceSDKManager`?  That's the current chain of cleanup events on logout.